### DISCUSSION TITLE
Clamps amperage calibration

### DISCUSSION THREAD

**Discussion by lucazeta1975** [JGAguado/Smart_Powermeter#35] [Show and tell]
After purchasing the Smart Powermeter, I created and downloaded the .bin firmware based on the .yaml listing proposed by Jon. I believe that every user, for simplicity, would do the same, and then proceed with their own customization (in fact, it can be noted that the total_daily_energy, for example, is given only by the sum of the values of the first three clamps. In my case, all the values of the 6 clamps will need to be summed to get an overall result).

The first thing I noticed, however, was that the Ampere values read were intuitively incorrect. This is due to the calibration filters set for each clamp.

So, I built a setup that allowed me to verify the electrical consumption values read by the Smart Powermeter and simultaneously by a calibrated multimeter, in order to obtain a plausible calibration of the six clamps provided.

![IMG_20240918_211047](https://github.com/user-attachments/assets/1cfc2740-7fe5-4310-934b-fe97d9bef151)

I proceeded by connecting an appliance (a hairdryer in my case) that allowed me to have various electrical consumption values.
I reprogrammed the Smart Powermeter by excluding all filters as correction coefficients of the read value, and then took photographs that included in the same frame the value read by the multimeter and the value read from the real-time log of Esphome.

![IMG_20240918_213752](https://github.com/user-attachments/assets/792ce8dd-dee6-4496-80cc-9ed9fb5992dd)

Based on these notes, I reconstructed a calibration table more plausible to reality. I noticed that all the clamps behave more or less in the same way. If there were a factory value that differs from one clamp to another, there could certainly be a bias, so it would be worth having a different calibration clamp by clamp, but given the domestic use, this bias becomes negligible and a “mean” reference of the read values can be taken.

This is my replacement parts of the list.
You can notice the filter calibration tables.

  - platform: ct_clamp
    sensor: Input_1
    id: Probe_1
    name: "Probe 1"
    sample_duration: 200ms
    update_interval: 5s
    filters:
     - calibrate_linear:
         - 0 -> 0
         - 0.002 -> 0.190
         - 0.0082 -> 0.824
         - 0.0153 -> 1.465
         - 0.0262 -> 2.513
         - 0.0555 -> 5.209
         - 0.0868 -> 8.320

    
  - platform: ct_clamp
    sensor: Input_2
    name: "Probe 2"
    id: Probe_2
    sample_duration: 200ms
    update_interval: 5s
    filters:
     - calibrate_linear:
         - 0 -> 0
         - 0.0079 -> 0.812
         - 0.0082 -> 0.818
         - 0.0140 -> 1.374
         - 0.0254 -> 2.484
         - 0.0867 -> 8.150
         
    
  - platform: ct_clamp
    sensor: Input_3
    name: "Probe 3"
    id: Probe_3
    sample_duration: 200ms
    update_interval: 5s
    filters:
      - calibrate_linear:
          - 0 -> 0
          - 0.0079 -> 0.812
          - 0.0137 -> 1.369
          - 0.0259 -> 2.478
          - 0.0578 -> 5.489
          - 0.0847 -> 8.200

  - platform: ct_clamp
    sensor: Input_4
    name: "Probe 4"
    id: Probe_4
    sample_duration: 200ms
    update_interval: 5s
    filters:
      - calibrate_linear:
          - 0 -> 0
          - 0.0079 -> 0.820
          - 0.0134 -> 1.378
          - 0.0263 -> 2.490
          - 0.0551 -> 5.162
          - 0.0567 -> 5.512
          - 0.0850 -> 8.280

  - platform: ct_clamp
    sensor: Input_5
    name: "Probe 5"
    id: Probe_5
    sample_duration: 200ms
    update_interval: 5s
    filters:
      - calibrate_linear:
          - 0 -> 0
          - 0.0083 -> 0.821
          - 0.0135 -> 1.384
          - 0.0257 -> 2.499
          - 0.0536 -> 5.182
          - 0.0575 -> 5.535
          - 0.0860 -> 8.250

  - platform: ct_clamp
    sensor: Input_6
    name: "Probe 6"
    id: Probe_6
    sample_duration: 200ms
    update_interval: 5s
    filters:
      - calibrate_linear:
          - 0 -> 0
          - 0.0081 -> 0.825
          - 0.0140 -> 1.392
          - 0.0265 -> 2.521
          - 0.0547 -> 5.221
          - 0.0577 -> 5.583
          - 0.0872 -> 8.330

**Comment by lucazeta1975** [comment]
I suggest you anyway to have a check of the values by yourself, if possible. Once mounted in mains electric panel, clamps may have a different "sensibility".
I will update this discussion soon with further documentation.

**Comment by vincegre** [comment]
Hi @lucazeta1975 

Thanks a lot for the share, I'm in the process myself to setup a Powermeter in my kitchen electrical panel. My calibrations settings are quite similar at yours ;)
May you explain why you changed the update_interval at 5s ?
Also did you notice any issues while everything closed back in electrical panel ? I have calibrated each of my CT by pulling cable a little apart for easy access and also install the multimeter but now everything is back packed in electrical panel I have some permanent quite stable consumption of 0.1A on circuits that should not have.

Thanks, best

Vincèn

**Reply by lucazeta1975** [reply]
Hello, what you describe is exactly what I have noticed as well. The clamp inserted in the electrical panel measures a current of 0.1A even without internal current flow. It is very likely that there may be external electromagnetic influences on the clamp itself.

One of my next tests will indeed be to calibrate the clamps already inserted in the electrical panel by interposing the multimeter directly on the various main switches, obviously in a safe manner.

Regarding the 5-second value, I set that value to avoid hypertrophic sensors sending almost useless amounts of data daily. On other smart plugs I own, I have set an update value of 10 seconds. That is enough to understand and record a reasonably significant electrical consumption

**Comment by lucazeta1975** [comment]
![IMG_20241002_231556](https://github.com/user-attachments/assets/34d8135e-9f0a-4c61-80cc-15eef0d771ac)

Today I conducted further investigations because I noticed strange electrical consumption on the line of my air conditioner (HVAC, two splitters).

Even though the air conditioners were off, I was seeing ~100W of consumption, which made me suspicious, fearing that it could be a design flaw in the clamps. However, with my multimeter, I discovered that the phantom consumption was real, and that the system in standby, even considering the various wireless connections, indeed consumes a certain amount of power.

By turning off the circuit breaker, I also discovered that the phantom currents detected by the clamp were real, but at different values.

I took a screenshot of the graphical log from one of the Smart Power Meter clamps, but it seems to me that the same behavior could be observed on all the clamps. In light of these observations, the calibration template needs to be reconsidered.

![Immagine 2024-10-02 234608](https://github.com/user-attachments/assets/db72def9-49d3-4180-b96d-05379f2636bc)

This is what I edited:

 - platform: ct_clamp
    sensor: Input_5
    name: "Probe 5"
    id: Probe_5
    sample_duration: 200ms
    update_interval: 5s
    filters:
      - calibrate_linear:
          - 0 -> 0
          - 0.0018 -> 0  #Phantom current zone not to log
          - 0.0049 -> 0.459
          - 0.0083 -> 0.821
          - 0.0135 -> 1.384
          - 0.0257 -> 2.499
          - 0.0536 -> 5.182
          - 0.0575 -> 5.535
          - 0.0860 -> 8.250
